[18.0][MIG] edi_oca

JordiMForgeFlow opened the pull request:
depends on:

- [x] https://github.com/OCA/edi/pull/1055

includes all FW in https://github.com/OCA/edi-framework/pull/125

etobella committed:
[FIX] edi: output might be not UTF-8 encodable, like PDFs

Co-authored-by: Núria Sancho <nuria.sancho@creublanca.es>

OCA-git-bot committed:
edi_oca 14.0.1.2.0

simahawk committed:
edi_backend: update docstring

simahawk committed:
edi_backend: fix var name

simahawk committed:
edi: exc record get file content as b64 optionally

simahawk committed:
edi: improve exchange record UI

simahawk committed:
edi: improve backend view extension

simahawk committed:
edi: allow job channel conf by type

simahawk committed:
edi: specific method to get type settings

simahawk committed:
edi: load main vars at init

simahawk committed:
edi_backend: exclude class name from error txt

simahawk committed:
edi: exclude exception class from error msg

simahawk committed:
edi: improve exchange type and record UI

simahawk committed:
edi: add 'retry' action to exchange record

This action allows to re-schedule the record for the action that failed
depending on its current state and direction.

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.3.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

etobella committed:
[IMP] edi: Allow to ignore some jobs

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.4.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

simahawk committed:
edi: fix wrong status bar display new exc record

Without this check you get 2 overlapping bars till you select an exc type.

simahawk committed:
edi: fix consumer mixin

Exchange type has no display_name

etobella committed:
[IMP] edi: ext identifier unique by type and backend

It might be possible to have the same external_identifier for different backends and types.

OCA-git-bot committed:
edi_oca 14.0.1.5.0

OCA-git-bot committed:
edi_oca 14.0.1.5.1

simahawk committed:
edi: backend._get_component inject custom ctx

You can now provide additional env context from type settings.

oca-travis committed:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.6.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

OCA-git-bot committed:
edi_oca 14.0.1.6.1

simahawk committed:
edi_oca: remove filename from status msg

The filename is redundant there.
It's pointless to see it every time in the chatter.
If you want to see the details, you go to the exchange record.

simahawk committed:
edi: chatter mgs tmpl fix link

Linking via 'oe' attrs seems not to work.
Having a real URL also allows to open the record in a new tab.

simahawk committed:
edi: exchange record state use badge widget in tree view

This makes the state more prominent and clear.

etobella committed:
[IMP] edi_oca: Pass development status to Beta

OCA-git-bot committed:
[UPD] README.rst

i-vyshnevska committed:
[MIG][15.0] edi_oca

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 14.0.1.7.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-14.0/edi-14.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-14-0/edi-14-0-edi_oca/

Commit:
[FIX] Fix access right issues when using actions

JordiMForgeFlow committed:
[FIX] edi_oca: fix chatter msg anchor

By adding the span element, the navigation is not performed correctly
and the anchor just redirects to blank '#'. Just remove the span.

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi_oca 15.0.1.0.1

simahawk committed:
edi: fix exc record name compute

No more ugly names in the form view till you populate all fields.

simahawk committed:
edi: improve exchanges views and menu

* add quick filters
* add quick menu items

simahawk committed:
edi: fix exc record missing _rec_name

simahawk committed:
edi: exc record form name not needed

simahawk committed:
edi: fix exc record sorting

simahawk committed:
edi: get rid of name, use identifier

Name contained redundant information like type name and related record.
On for and tree view is totally useless.
To still provide information on linked resource name
the 'related_name' field has been added.

simahawk committed:
edi: backend action to view exchanges

simahawk committed:
edi: backend action to view exchange types

simahawk committed:
edi: move job func to exc record

Prior to this change jobs were tied to the backend.

The backend must be responsible for processing records
and this is why the logic has been implemented there
and it's kept there with this change too.

However, it's way better if the job is tied to its record directly
because it makes more sense and it helps jumping to the record
w/ the 'related' btn on the job.

simahawk committed:
edi: update exc type adv settings help

Commit:
[UPD] Update edi_oca.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-15.0/edi-15.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-15-0/edi-15-0-edi_oca/

simahawk committed:
edi: show exchanges first

Makes little sense to show backends first
when you enter the EDI menu.
Ongoing exchanges are the most important thing.

simahawk committed:
edi_oca: fix exchange chatter link

simahawk committed:
edi_oca: avoid warning on exception args

etobella committed:
[FIX] edi_oca: Restore removed functions

We want to keep functionality on the same version. It is marked to remove for 16

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 15.0.1.1.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-15.0/edi-15.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-15-0/edi-15-0-edi_oca/

JordiMForgeFlow committed:
[IMP] edi_oca: add exchange_filename search

jcoux committed:
edi_oca: allow to override dt for exchange filename

By setting specific timezone and datetime format on exchange type
advanced settings, we can easily override the datetime used for generation of
exchange record filename.

OCA-git-bot committed:
edi_oca 15.0.1.2.0

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
edi_oca 15.0.1.3.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-15.0/edi-15.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-15-0/edi-15-0-edi_oca/

JaumeBforgeFlow committed:
[IMP] edi: archive edi_backend

simahawk committed:
edi: silent loggers in tests

simahawk committed:
edi_oca: add more generic mixin

Not all components might be bound to an exchange record.

simahawk committed:
edi_oca: add some TODOs

simahawk committed:
edi_oca: improve auto btns

* auto btns on/off flag per exchange type
* offer hook to customize how btns are rendered
* improve UI

simahawk committed:
edi_oca: update ownership

simahawk committed:
edi_oca: fix docstring on consumer mixin

simahawk committed:
edi: add meth _is_valid_edi_action

Handy method to validate core actions of the edi framework.

simahawk committed:
edi: improve consumer mixin helpers

simahawk committed:
edi: fix exc_rec.action_exchange_generate

Kewyord params were not propagated.

simahawk committed:
edi: improve exc_rec._trigger_edi_event

Propagate custom kw args to event handlers.

simahawk committed:
edi: fix process break on error ctx key

simahawk committed:
edi: improve create record validation

Raise specific error when exchange type is not found.

simahawk committed:
edi: ease creation of child records

simahawk committed:
edi: add edi_id_mixin

To be used to track references on exchanged related records.

simahawk committed:
edi_oca: bump 15.0.1.4.0

simahawk committed:
edi: fix consumer mixin test

simahawk committed:
edi: improve consumer mixin w/ origin

You can now track the original exchange record
that originated the document in the 1st place.

simahawk committed:
edi: add test for create child/ack

simahawk committed:
edi: add 'ack for' on type

Ease finding which type the current one is an ack for.

simahawk committed:
edi: improve exc type search

simahawk committed:
edi: fix ack record compute

You can produce or receive more than one ack for a given exchange.
If that's the case, make sure only the newest is considered.

simahawk committed:
edi: fix exchange ordering

In case exchange_on is the same, use ID to get the latest on top.

simahawk committed:
edi: fix record and type copy on fields

simahawk committed:
edi: fix error msg typo

simahawk committed:
edi: add exc.type.set_settings method

simahawk committed:
edi: improve chatter msg w/ type detail

simahawk committed:
edi: fix _cron_check_output_exchange_sync

Skip 'output_sent' records by default.
Reason: most of the times when a message is sent out
you don't care about its state anymore.
Any subsequent update from outside should come
with a new input record instead.

Yet, if for any reason this feature is required
you can still enable it by passing 'skip_sent=False'.

ivantodorovich committed:
[IMP] edi_oca: bump icon to 15.0 style

OCA-git-bot committed:
edi_oca 15.0.1.3.1

simahawk committed:
edi: allow search consumers by exc type

simahawk committed:
edi: fix action_view_edi_records

As we are reusing the existing window action
we must remove default filters
otherwise some related exchange records will be hidden by default.

simahawk committed:
edi: improve consumer record count perf

simahawk committed:
edi: fix record missing indexes

simahawk committed:
edi: add test for exc.record.action_retry

simahawk committed:
edi: fix exc.record._compute_retryable

simahawk committed:
edi: imp backend check input/output

Accept `record_ids` to work w/ specific records.

simahawk committed:
edi: improve cron names

simahawk committed:
edi: add quick exec option by type

Allow scheduling jobs immediately based on type configuration.
Very often you want an action to be executed right away.
You still get a job, meaning that the action is not sync
but the job will be scheduled immediately
instead of waiting for the cron to run.

simahawk committed:
edi: exchange type add TODO

simahawk committed:
edi: fix backward compat usage of _has_exchange_record_domain

In this commit https://github.com/OCA/edi/commit/0a9b6d5ef18f9d6c7d493ebefee8e579c697a165
I unified the way these kind of methods were used.

Was breaking older implementations where the method was called directly.

simahawk committed:
edi: drop ack record auto create

Creating the ack was kind of... an hack :)

When I added this behavior we didn't have yet the storage machinery
that looks automatically for a new file.
Also for webservices seems not relevant since we can configure endpoints
that can create the ack when needed.

All in all, I don't think this feature is needed.
Worse, it adds some overhead when a record gets created
and can generate write conflicts when exchanges are processed immediately
overlapping with the creation of the ack record.

simahawk committed:
edi: add 'as_bytes' option to _get_file_content

When sending the file data, we might want to use its bytes version.
This new flag gives an option for it.

simahawk committed:
edi: fix _get_component ctx propagation

The env_ctx that might come from a exchange type settings
was not propagated to the exchange record
which is made directly available to the work ctx of the component.

Before this change, if you had any method on the exchange record
or on the related record relying on a ctx key,
the ctx key was not propagated to all records' context.

victoralmau committed:
[FIX] edi_oca: Test errors

Co-authored-by: etobella

OCA-git-bot committed:
edi_oca 15.0.1.3.2

OCA-git-bot committed:
edi_oca 15.0.1.3.3

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
edi_oca 15.0.1.5.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-15.0/edi-15.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-15-0/edi-15-0-edi_oca/

john-herholz-dt committed:
[MIG] edi_oca: Migration to 16.0

simahawk committed:
edi_oca: fix edi_id label

florentx committed:
edi_oca: avoid ghost exchange records

florentx committed:
edi_oca: avoid backend mismatch

florentx committed:
edi_oca: assertion was not executed

florentx committed:
edi_oca: log warning for deleted records

simahawk committed:
edi: unify action complete notification

Now all main framework actions call the same notification method when complete.

simahawk committed:
edi: trigger generic event on action complete

simahawk committed:
edi: mark every session w/ 'edi_framework' ctx key

simahawk committed:
edi: trigger generic event on related record

simahawk committed:
edi: exc.record._set_related_record use sudo

Motivation: you end up using sudo on every call
because you don't know if the current user will have perm.
Yet, we don't care for such operatation.

simahawk committed:
edi: speed up exchange record views

* filter by default on the last 15 days w/ 'Recent' menu item
* filter by default on today on all menu items
* allow to browse them all w/ 'All' at the bottom

simahawk committed:
edi: add 'active' field to exchange type

simahawk committed:
edi: improve backend view filter inactive

simahawk committed:
edi: add disable flag to consumer

simahawk committed:
edi: allow enable/disable exc type by partner

simahawk committed:
edi: fix related record missing broken views

Before this change, when a related record was deleted
you could not access anymore any exchange record (tree or form)
due to the permission check.

Now:

* the check is bypassed
* related record buttons are hidden
* an informative message is displayed in the form

petrus-v committed:
edi_oca: fix consumer mixin exchange record count

simahawk committed:
edi: refactor model rules w/ specific model

Long awaited improvement.
Allows to decouple rules by model giving much more flexibility.
Migration steps included.

simahawk committed:
edi: unify menu item labels

simahawk committed:
edi: allow form btn actions customization

You can now customize and translate labels and tooltips for EDI automatic buttons.

simahawk committed:
edi: add generate_ok message

When actions are automated, is nice to see that data has been generated for an exchange on the related record.

simahawk committed:
edi: chain generate/send jobs

Instead of waiting for the cron to pass again and send the file
chain the 2 jobs so that it gets sent right after generation.

simahawk committed:
edi: fix backend jobs test old api

The recommended way to execute actions on records
is to call `action_exchange_*` on

simahawk committed:
edi_oca: tests.common ease ctx setup

simahawk committed:
edi: refactor model rules backward compat

simahawk committed:
edi_oca: fix test class names

Having the same name for different classes prevents selection of
specific test cases when running single test suites.

simahawk committed:
edi_oca: fix edi.exchange.type.create

simahawk committed:
edi_oca: move to OCA/edi-framework

simahawk committed:
edi_oca: use queue_job__no_delay key in tests

simahawk committed:
edi_oca: drop migrations

simahawk committed:
edi_oca: drop deprecated code

simahawk committed:
edi_oca: make not installable

petrus-v committed:
edi_oca: avoid duplicate field name

Avoid warning because of duplicated Model field name while installing this module

simahawk committed:
edi: exc type fix view archived rules

Archived rules were not visible on the exc type form.

simahawk committed:
edi: exc type view allow archive rules

You can now toggle archived/unarchived directly from exc type form

simahawk committed:
edi: fix tests

simahawk committed:
edi: rename consumer mixin disable_edi_auto field

nilshamerlinck committed:
edi_oca: fix _logger.debug usage

TypeError: not all arguments converted during string formatting

simahawk committed:
edi: fix access errors tests

AccessError core messages were modified

simahawk committed:
edi: drop useless view form/tree strings

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

simahawk committed:
edi: fix consumer get_view w/ nested forms

Select main  only as they can be nested into field custom forms.
I'm looking at you  on v16 :S

OCA-git-bot committed:
[BOT] post-merge updates

etobella committed:
[FIX] edi_oca: Show buttons properly. Use OWL

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update edi_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-16.0/edi-framework-16.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-16-0/edi-framework-16-0-edi_oca/

simahawk committed:
edi: fix job return msg

simahawk committed:
edi: make send failed job retryable

Send jobs might fail due to an external service being not responsive.
If the job is simply failed, a new one will be spawned and might encour in the same error again,
possibly leading to an high number of duplicated jobs for the same record.

Yet, when a RetryableJobError is raised, the job will be set back into pending state
and will be nicely retried based on jobs configuration.

OriolMForgeFlow committed:
[IMP] edi_oca: add company_id field with multi-company rule

MiquelRForgeFlow committed:
[FIX] edi_oca: avoid flooding error in the record chatter

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[UPD] Update edi_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-16.0/edi-framework-16.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-16-0/edi-framework-16-0-edi_oca/

HviorForgeFlow committed:
[IMP] edi_oca: Show traceback on exception errors

OCA-git-bot committed:
[BOT] post-merge updates

simahawk committed:
edi: test_same_code_same_backend mute sql logger

simahawk committed:
edi: improve send job retry

* catch OSError
* add debug log

simahawk committed:
edi: add file checksum

simahawk committed:
edi: use job identity_key

Prevents having duplicated jobs for the same record as far as possible.

simahawk committed:
edi: raise send job prio to max

Try to send out the file as fast as possible once the content is ready.

simahawk committed:
edi: fix consumer mixin copy origin exc record

The origin record should never been copied.
Its values should always be set specifically by a framework action.

Commit:
[IMP] edi_oca: support exchange filename without extension on exchange record

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[IMP] edi_oca: add partner form page

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-16.0/edi-framework-16.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-16-0/edi-framework-16-0-edi_oca/

nilshamerlinck committed:
[FIX] edi_oca: tests using FakeModelLoader should not be post_install

OCA-git-bot committed:
[BOT] post-merge updates

tuantrantg committed:
[IMP] edi_oca: allow use of sequence for exchange filename pattern

JordiMForgeFlow committed:
[IMP] edi_oca: add encoding management for exchange type

john-herholz-dt committed:
[IMP] edi_oca: pre-commit auto fixes

john-herholz-dt committed:
[MIG] edi_oca: Migration to 17.0

Commit:
[FIX] edi_oca: fixs after porting from 14.0

Commit:
edi: fix archive rules when type archived

Commit:
edi_oca: remove warnings in tests

JordiMForgeFlow committed:
[IMP] edi_oca: edi_backend multi-company

gurneyalex committed:
[IMP] edi_oca: add a backend_type_code field on the backend

this field can be used in backend views to show/hide configuration elements depending on the backend type

Commit:
[IMP] edi_oca: Add Retry action on edi.exchange.record list view

florentx committed:
edi_oca: Fix Access error for Administrator

Commit:
[IMP] edi_oca: allow to receive empty files

Currently when we receive an empty file:

- Receive step doesn't end up in state error_on_receive : that's because we just ignore the content here
- But the process step will raise an error
- So this commit goal tries to improve that
- By default, this option is disabled so we still consider that an empty file is an error case

JordiMForgeFlow committed:
[IMP] edi_oca: add button to regenerate file after error on validation

nilshamerlinck committed:
[FIX] edi_oca: do not try to automatically re-process input_processed_error records

HviorForgeFlow committed:
[FIX] edi_oca: Don't run quick_exec if backend is not active

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

Commit:
[IMP] edi_oca: support modifying failed check send message

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (202 of 202 strings)

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/it/

Commit:
[IMP] edi_oca: update docs for EDI Jobs

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (202 of 202 strings)

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/it/

Commit:
[UPD] Update edi_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/

QuocDuong1306 committed:
[IMP] edi_oca: add btn to show all related queue jobs

Commit:
[UPD] Update edi_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (204 of 204 strings)

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/it/

MiquelRForgeFlow committed:
[FIX] edi_oca: improve error in order to be helpful

manuelregidor committed:
[FIX]edi_oca: error in component

OCA-git-bot committed:
[BOT] post-merge updates

HviorForgeFlow committed:
[IMP] edi_oca: Add hook on new output exchange record search

etobella committed:
[FIX] edi_oca: Function for posting messages was renamed

OCA-git-bot committed:
[BOT] post-merge updates

JordiMForgeFlow committed:
[IMP] edi_oca: pre-commit stuff

JordiMForgeFlow committed:
[IMP] edi_oca: adapt FWs to V17

JordiMForgeFlow committed:
[MIG] edi_oca: Migration to V18

JordiMForgeFlow committed:
[ADD] test-requirements

JordiMForgeFlow commented:
@simahawk @etobella 

The CI fails due to the warning logged in https://github.com/OCA/edi-framework/blob/03961fbed1c6269153c56fa038e967300904d28c/edi_oca/models/edi_exchange_record.py#L490

There is one test that is also ensuring that the warning is logged.

How would you deal with this one? Maybe use an info log when running the tests?

thienvh332 commented on edi_oca/tests/test_security.py:
```diff
@@ -0,0 +1,223 @@
+        exchange_record.res_id = -1
+        admin_group = self.env.ref("base.group_system")
+        self.user.write({"groups_id": [(4, self.group.id), (4, admin_group.id)]})
+        self.assertEqual(
+            1,
+            self.env["edi.exchange.record"]
+            .with_user(self.user)
+            .search_count([("id", "=", exchange_record.id)]),
+        )
```
Changing this will solve the problem you are having. :wink:
```suggestion
        logger_name = "odoo.addons.edi_oca.models.edi_exchange_record"
        with self.assertLogs(logger_name, "WARNING"):
            self.assertEqual(
                1,
                self.env["edi.exchange.record"]
                .with_user(self.user)
                .search_count([("id", "=", exchange_record.id)]),
            )
```

JordiMForgeFlow replied:
Thank you @thienvh332 🙌🏼

JordiMForgeFlow commented:
@etobella @simahawk ready for review :)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (204 of 204 strings)

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (204 of 204 strings)

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/it/

SilvioC2C committed:
[IMP] edi_oca: remove logs' warnings causes

- remove usage of ``active_id`` from view
- de-duplicate fields' strings

OCA-git-bot committed:
[BOT] post-merge updates

SilvioC2C committed:
[FIX] edi_oca: adapt forward-ported commits to v17.0

Commit:
[UPD] Update edi_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (226 of 226 strings)

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/it/

Commit:
[UPD] Update edi_oca.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (227 of 227 strings)

Translation: edi-framework-17.0/edi-framework-17.0-edi_oca
Translate-URL: https://translation.odoo-community.org/projects/edi-framework-17-0/edi-framework-17-0-edi_oca/it/

MiquelRForgeFlow commented:
I think you should consider as base https://github.com/OCA/edi-framework/pull/125 instead of https://github.com/OCA/edi-framework/pull/118, and then include any missing FW commit.

JordiMForgeFlow committed:
[IMP] edi_oca: pre-commit stuff

JordiMForgeFlow commented:
@MiquelRForgeFlow done!

thienvh332 commented on edi_oca/views/edi_exchange_record_views.xml:
```diff
@@ -0,0 +1,400 @@
+        <field name="model">edi.exchange.record</field>
+        <field name="arch" type="xml">
+            <form string="EDI Exchange Record">
+                <field name="direction" invisible="1" />
+                <field name="retryable" invisible="1" />
+                <field name="related_record_exists" invisible="1" />
```
FYI, required field (invisible) added automatically in views. (See `https://github.com/odoo/odoo/pull/137031`)

Other similar places

```suggestion

```

JordiMForgeFlow replied:
@thienvh332 done :)

thienvh332 reviewed (approved):
LG overall,

Small changes:

JordiMForgeFlow committed:
[IMP] edi_oca: remove required fields from view

https://github.com/odoo/odoo/pull/137031

JordiMForgeFlow commented:
@etobella ready to merge :)

etobella commented:
/ocabot merge nobump
